src/shared/resourcefetcher/compositeResourceFetcher.ts

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getLogger, Logger } from '../logger'
+import { ResourceFetcher } from './resourcefetcher'
+
+export class CompositeResourceFetcher implements ResourceFetcher {
```

Comment:
I like this strategy much better than the 'fallback' style we have in jetbrains - much more flexible - we should port this across to the JB implementation.